src/thunderbots/software/network_input/backend.cpp

```diff
@@ -132,7 +132,17 @@ Ball Backend::getFilteredBallData(const std::vector<SSL_DetectionFrame> &detecti
             ball_detection.position =
                 Point(ball.x() * METERS_PER_MILLIMETER, ball.y() * METERS_PER_MILLIMETER);
             ball_detection.timestamp = Timestamp::fromSeconds(detection.t_capture());
-            ball_detections.push_back(ball_detection);
+
+            bool ignore_ball =
+                Util::DynamicParameters::AI::refbox::ignore_camera_data.value() &&
+                (Util::DynamicParameters::AI::refbox::min_valid_x.value() >
+                     ball_detection.position.x() ||
+                 Util::DynamicParameters::AI::refbox::max_valid_x.value() <
+                     ball_detection.position.x());
```

Comment:
Could we change this to: 
```
            bool ball_position_invalid = (Util::DynamicParameters::AI::refbox::min_valid_x.value() >
                     ball_detection.position.x() ||
                 Util::DynamicParameters::AI::refbox::max_valid_x.value() <
                     ball_detection.position.x());
            bool ignore_ball = Util::DynamicParameters::AI::refbox::ignore_camera_data.value() && ball_position_invalid;
```

Reply:
Also, could we change `ignore_camera_data` to `ignore_invalid_camera_data`?

Reply:
And as a thought, shouldn't we always be ignoring invalid camera data?

Reply:
Yes, but this is a "custom" version of invalid because somethings we want to artifically constrict the data we get. From a UX persepctive, it's easier to have a toggle to enable and disable this in addition to defining the range of the valid data.

Reply:
I'm confused, `ball_position_invalid` would mean just that to me?  What's the case where `ball_position_invalid` is true and we still want to use it? If that's the case, wouldn't the ball position be be valid?

Reply:
In my head there's a distinction between a ball position being generally invalid (eg. it's way outside the field boundaries) vs "artificially" invalid (when we artificially restrict the "valid" area).

The `ignore_camera_data` param toggles whether or not we want to use the "artificial" invalid restrictions in addition to the regular ones. It is useful to have this toggle separate from the `ball_position_invalid` because then we can clearly enable or disable the extra restrictions, without having to constantly change the "artificial" boundaries to match what we want. Imagine wanting to quickly turn on and off these restrictions (maybe for some vision testing or something), it's much easier with a separate parameter that enables the restrictions.

This does mean that `ball_position_invalid` can be true but we still use the ball position.

If this still isn't clear lets talk in person.

Reply:
@garethellis0 	is this good?